api/environments/identities/serializers.py

```diff
@@ -66,11 +68,28 @@ class _TraitSerializer(serializers.Serializer):  # type: ignore[type-arg]
 
 class SDKIdentitiesQuerySerializer(serializers.ModelSerializer[Identity]):
     transient = serializers.BooleanField(default=False)
+    identifier = serializers.CharField(required=True, max_length=2000)
 
     class Meta:
         model = Identity
         fields = ("identifier", "transient")
 
+    def validate_identifier(self, value: str) -> str:
+        try:
+            identifier_regex_validator(value)
+            return value
+        except DjangoValidationError:
+            request = self.context.get("request")
+            environment = getattr(request, "environment", None)
+            if (
+                environment
+                and Identity.objects.filter(
+                    environment=environment, identifier=value
+                ).exists()
+            ):
+                return value
+            raise ValidationError(identifier_regex_validator.message)
+
```

Comment:
Once the validation rule is fixed in order to allow white spaces, this should not be necessary.